src/datumaro/components/prune.py

```diff
@@ -0,0 +1,322 @@
+        ratio, len(database_keys), cluster_num_item_list
+    )
+
+    selected_items = []
+    dist_tuples = []
+    for i, cluster_id in enumerate(cluster_ids):
+        cluster_center = cluster_centers[cluster_id]
+        cluster_items_idx = np.where(clusters == cluster_id)[0]
+        num_selected_item = norm_cluster_num_item_list[i]
+
+        cluster_items = database_keys[cluster_items_idx]
+        dist = calculate_hamming(cluster_center, cluster_items)
+        ind = np.argsort(dist)
+        item_idx_list = cluster_items_idx[ind]
+        for i, idx in enumerate(item_idx_list[:num_selected_item]):
+            selected_items.append(item_list[idx])
+            dist_tuples.append((cluster_id, item_list[idx].id, item_list[idx].subset, dist[ind][i]))
+    return selected_items, dist_tuples
+
+
+def entropy(ratio, num_centers, database_keys, labels, item_list, source):
+    """
+    Select items through clustering and choose them based on label entropy in each cluster.
+    """
+    kmeans = KMeans(n_clusters=num_centers, random_state=0)
+    clusters = kmeans.fit_predict(database_keys)
+
+    cluster_ids, cluster_num_item_list = np.unique(clusters, return_counts=True)
+    norm_cluster_num_item_list = match_num_item_for_cluster(
+        ratio, len(database_keys), cluster_num_item_list
+    )
+
+    selected_item_indexes = []
+    for cluster_id, num_selected_item in zip(cluster_ids, norm_cluster_num_item_list):
+        cluster_items_idx = np.where(clusters == cluster_id)[0]
+
+        cluster_classes = np.array(labels)[cluster_items_idx]
+        _, inv, cnts = np.unique(cluster_classes, return_inverse=True, return_counts=True)
+        weights = 1 / cnts
+        probs = weights[inv]
+        probs /= probs.sum()
+
+        choices = np.random.choice(len(inv), size=num_selected_item, p=probs, replace=False)
+        selected_item_indexes.extend(cluster_items_idx[choices])
+
+    selected_items = np.array(item_list)[selected_item_indexes].tolist()
+    return selected_items, None
+
+
+def ndr_select(ratio, num_centers, database_keys, labels, item_list, source):
+    """
+    Select items based on NDR among each subset.
+    """
+    subset_lists = list(source.subsets().keys())
+
+    selected_items = []
+    for subset_ in subset_lists:
+        subset_len = len(source.get_subset(subset_))
+        num_selected_subset_item = math.ceil(subset_len * (1 - ratio))
+        ndr_result = ndr.NDR(source, working_subset=subset_, num_cut=num_selected_subset_item)
+        selected_items.extend(ndr_result.get_subset(subset_))
+
+    return selected_items, None
+
+
+class Prune:
+    def __init__(
+        self,
+        *datasets: Dataset,
+        cluster_method: str = "random",
+        hash_type: str = "img",
+    ) -> None:
+        """
+        Prune make a representative and manageable subset.
+        """
+        self._dataset = datasets[0] if isinstance(datasets, tuple) else datasets[0]
```

Comment:
Why do you take a sequence of datasets in L204 as the input argument if you want to store just one of them?

Reply:
Fix to get one dataset.